src/closures/mixing_length.jl

```diff
@@ -0,0 +1,78 @@
+function mixing_length(param_set, ml_model::MinDisspLen)
+    l = zeros(3)
+    c_m = CPEDMF.c_m(param_set)
+    c_d = CPEDMF.c_d(param_set)
+    c_b = ICP.static_stab_coeff(param_set)
+    g = CPP.grav(param_set)
+
+    # kz scale (surface layer)
+    if ml_model.obukhov_length < 0.0 #unstable
+        l_W =
+            ml_model.κ_vk * ml_model.z / (sqrt(ml_model.tke_surf / ml_model.ustar / ml_model.ustar) * c_m) *
+            min((1.0 - 100.0 * ml_model.z / ml_model.obukhov_length)^0.2, 1.0 / ml_model.κ_vk)
+    else # neutral or stable
+        l_W = ml_model.κ_vk * ml_model.z / (sqrt(ml_model.tke_surf / ml_model.ustar / ml_model.ustar) * c_m)
+    end
+
+    # compute l_TKE - Production/destruction terms
+    a_pd =
+        c_m * (ml_model.Shear² - ml_model.∂b∂z_θl / ml_model.Pr - ml_model.∂b∂z_qt / ml_model.Pr) * sqrt(ml_model.tke)
+    # Dissipation term
+    c_neg = c_d * ml_model.tke * sqrt(ml_model.tke)
+    # Subdomain exchange term
+    b_exch = 0.0
+    for nn in xrange(ml_model.N_up)
+        wc_upd_nn = ml_model.wc_up[nn]
+        wc_env = ml_model.wc_en
+        b_exch +=
+            ml_model.a_up[nn] * wc_upd_nn * ml_model.δ_dyn[nn] / ml_model.a_en *
+            ((wc_upd_nn - wc_env) * (wc_upd_nn - wc_env) / 2.0 - ml_model.tke) -
+            ml_model.a_up[nn] * wc_upd_nn * (wc_upd_nn - wc_env) * ml_model.ε_turb[nn] * wc_env / ml_model.a_en
+    end
+
+    l_entdet = 0.0 #-- IGNACIO without this the simulations break 
+    if abs(a_pd) > ml_model.ϵ0 && 4.0 * a_pd * c_neg > -b_exch * b_exch
+        l_entdet = max(-b_exch / 2.0 / a_pd + sqrt(b_exch * b_exch + 4.0 * a_pd * c_neg) / 2.0 / a_pd, 0.0)
+    elseif abs(a_pd) < ml_model.ϵ0 && abs(b_exch) > ml_model.ϵ0
+        l_entdet = c_neg / b_exch
+    end
+    l_TKE = l_entdet
+
+    # compute l_N²
+    # Effective static stability using environmental mean.
+    # Set lambda for now to environmental cloud_fraction (TBD: Rain)
+    N²_eff =
+        (1.0 - ml_model.en_cld_frac) * ml_model.∂θv∂z +
+        ml_model.en_cld_frac * (
+            1.0 / exp(-latent_heat(ml_model.T_en) * ml_model.ql_en / cpm_c(ml_model.qt_en) / ml_model.T_en) * (
+                (1.0 + (eps_vi - 1.0) * ml_model.qt_en) * ml_model.∂θl∂z +
+                (eps_vi - 1.0) * ml_model.θ_li_en * ml_model.∂qt∂z
+            )
+        )
+
+    N² = sqrt(max(g / ml_model.θv * N²_eff, 0.0))
+    if N² > 0.0
+        l_N² = min(sqrt(max(c_b * ml_model.tke, 0.0)) / N², 1.0e6)
+    else
+        l_N² = 1.0e6
+    end
+
+    l[1] = l_N²
+    l[2] = l_TKE
+    l[3] = l_W
+
+    # add limiters 
+    j = 1
+    while (j <= length(l))
+        if l[j] < ml_model.ϵ0 || l[j] > 1.0e6
+            l[j] = 1.0e6
+        end
+        j += 1
+    end
+
+    # get soft minimum
+    min_len_num = argmin(l)
+    mixing_length = lamb_smooth_minimum(l, 0.1, 1.5)
+    ml_ratio = mixing_length / l[Int(min_len_num)]
+    return MixLen(min_len_num, mixing_length, ml_ratio)
```

Comment:
use `findmin` to get both minimum value and its index.
```suggestion
    min_len, min_len_ind = findmin(l)
    mixing_length = lamb_smooth_minimum(l, 0.1, 1.5)
    ml_ratio = mixing_length / min_len
    return MixLen(min_len_ind, mixing_length, ml_ratio)
```